util/src/builder/callback_data.rs

```diff
@@ -0,0 +1,175 @@
+use twilight_model::{
+    application::{callback::CallbackData, component::Component},
+    channel::{
+        embed::Embed,
+        message::{AllowedMentions, MessageFlags},
+    },
+};
+
+/// Builder for [`CallbackData`].
+///
+/// # Example
+/// ```
+/// use twilight_util::builder::CallbackDataBuilder;
+/// use twilight_model::channel::message::MessageFlags;
+///
+/// let callback_data = CallbackDataBuilder::new()
+///     .content("My callback message")
+///     .flags(MessageFlags::EPHEMERAL)
+///     .build();
+/// ```
+#[derive(Debug)]
+#[must_use = "builders have no effect if unused"]
+pub struct CallbackDataBuilder(CallbackData);
+
+impl CallbackDataBuilder {
+    /// Create a new builder to construct a [`CallbackData`].
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Set allowed mentions in the interaction response.
+    #[allow(clippy::missing_const_for_fn)]
+    pub fn allowed_mentions(mut self, allowed_mentions: AllowedMentions) -> Self {
+        self.0.allowed_mentions = Some(allowed_mentions);
+
+        self
+    }
+
+    /// Add a message component.
+    ///
+    /// Multiple component can be set by calling this method multiple times.
+    pub fn component(mut self, component: Component) -> Self {
+        match self.0.components {
+            Some(ref mut components) => components.push(component),
+            None => self.0.components = Some(vec![component]),
+        };
+
+        self
+    }
+
+    /// Set the content.
+    pub fn content(self, content: impl Into<String>) -> Self {
```

Comment:
let's take a signature of `content: String` here so we can make the function `const`

Reply:
The code fails to compile with a `const fn`:

```rs
pub const fn content(mut self, content: String) -> Self {
    self.0.content = Some(content);

    self
}
```

```
error[E0493]: destructors cannot be evaluated at compile-time
  --> util/src/builder/callback_data.rs:53:9
   |
53 |         self.0.content = Some(content);
   |         ^^^^^^^^^^^^^^
   |         |
   |         constant functions cannot evaluate destructors
   |         value is dropped here
   ```

Reply:
Right, I was conflating `String` with `&'a str`, my bad. We can still take a `String` here though